### PR TITLE
Update test_Format.m [complete test]

### PR DESCRIPTION
**Pull request by egolol**
Add tests:
- [x] %% Test 2.IL: Check ITEMLIST
- [ ] %% Test 2.DI: Check IDICT
- [x] %% Test 2.LL: Check CELL

### CODE REVIEW

**Review by annacanal — APPROVED** [review]

**Review by c-yuwei — APPROVED** [review]

**Review by c-yuwei — APPROVED** [review]

File: braph2genesis/src/ds/test_Format.m [hunk 1]
```diff
@@ -226,57 +226,25 @@
...
-% % %     [BRAPH2.STR ':Format:' BRAPH2.WRONG_OUTPUT], ...
-% % %     'Format.checkFormat(Format.ITEMLIST) does not return correct output.')
-% % % assert(~Format.checkFormat(Format.ITEMLIST, true), ...
-% % %     [BRAPH2.STR ':Format:' BRAPH2.WRONG_OUTPUT], ...
-% % %     'Format.checkFormat(Format.ITEMLIST) does not return correct output.')
-% % % assert(~Format.checkFormat(Format.ITEMLIST, 'string'), ...
-% % %     [BRAPH2.STR ':Format:' BRAPH2.WRONG_OUTPUT], ...
-% % %     'Format.checkFormat(Format.ITEMLIST) does not return correct output.')
+clear value
+clear element_class_list
```
Comment (giovannivolpe) [inline braph2genesis/src/ds/test_Format.m:230]: why do you need to clear this?

File: braph2genesis/src/ds/test_Format.m [hunk 2]
```diff
@@ -226,57 +226,25 @@
...
+clear wrong_value
+
+element_class_list{1} = Element();
+element_class_list{2} = Element();
+
```
Comment (giovannivolpe) [inline braph2genesis/src/ds/test_Format.m:235]: You should also add the good cases, given by elements obtained from itemlist.

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

File: braph2genesis/src/ds/test_Format.m (hunk 2 above)
Comment (egolol) [reply braph2genesis/src/ds/test_Format.m:236]: I don't understand this, if you want items defined in Element, do I uncomment Element functions to do this? For example getElementList. Or, as I am doing, defining an array of pointers of items of Element subclasses.

File: braph2genesis/src/ds/test_Format.m [hunk 3]
```diff
@@ -226,57 +226,26 @@
...
+element_item_list = cellfun(@(x) eval([x '()']), element_item_list_names, 'UniformOutput', false);
+
+% ITEMLIST formats that should NOT be accepted
+wrong_value{1} = 3.14;
+wrong_value{2} = true;
+wrong_value{3} = 'String';
+wrong_value{4} = {'1', '2', '3'};
+
+% tests
+Format.checkFormat(Format.ITEMLIST, element_item_list)
```
Comment (giovannivolpe) [inline braph2genesis/src/ds/test_Format.m:243]: This should be a for cycle over all possible correct values

File: braph2genesis/src/ds/test_Format.m [hunk 4]
```diff
@@ -226,57 +226,26 @@
...
-% % %     'Format.checkFormat(Format.ITEMLIST) does not return correct output.')
-% % % assert(~Format.checkFormat(Format.ITEMLIST, 'string'), ...
-% % %     [BRAPH2.STR ':Format:' BRAPH2.WRONG_OUTPUT], ...
-% % %     'Format.checkFormat(Format.ITEMLIST) does not return correct output.')
+clear value
+clear wrong_value
+
+% ITEMLIST formats that should be accepted
+element_item_list_names = subclasses('Element', [], [], true);
+element_item_list = cellfun(@(x) eval([x '()']), element_item_list_names, 'UniformOutput', false);
```
Comment (giovannivolpe) [inline braph2genesis/src/ds/test_Format.m:234]: make "element_item_list" as value
addd more cases, such as a list of {'Element'}, ... taken with a for cycle from element_list

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

**Review by giovannivolpe — COMMENTED** [review]
Let's double-check and revise it together when we speak

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**